vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MessageVoiceBroadcastRecordingItem.kt

```diff
@@ -0,0 +1,137 @@
+import im.vector.app.core.resources.ColorProvider
+import im.vector.app.core.resources.DrawableProvider
+import im.vector.app.features.home.room.detail.RoomDetailAction.VoiceBroadcastAction
+import im.vector.app.features.home.room.detail.timeline.TimelineEventController
+import im.vector.app.features.voicebroadcast.VoiceBroadcastRecorder
+import org.matrix.android.sdk.api.util.MatrixItem
+
+@EpoxyModelClass
+abstract class MessageVoiceBroadcastRecordingItem : AbsMessageItem<MessageVoiceBroadcastRecordingItem.Holder>() {
+
+    @EpoxyAttribute
+    var callback: TimelineEventController.Callback? = null
+
+    @EpoxyAttribute
+    var voiceBroadcastRecorder: VoiceBroadcastRecorder? = null
+
+    @EpoxyAttribute
+    lateinit var colorProvider: ColorProvider
+
+    @EpoxyAttribute
+    lateinit var drawableProvider: DrawableProvider
+
+    @EpoxyAttribute
+    var roomItem: MatrixItem? = null
+
+    @EpoxyAttribute
+    var title: String? = null
+
+    private lateinit var recorderListener: VoiceBroadcastRecorder.Listener
+
+    override fun isCacheable(): Boolean = false
+
+    override fun bind(holder: Holder) {
+        super.bind(holder)
+        bindVoiceBroadcastItem(holder)
+    }
+
+    private fun bindVoiceBroadcastItem(holder: Holder) {
+        recorderListener = object : VoiceBroadcastRecorder.Listener {
+            override fun onStateUpdated(state: VoiceBroadcastRecorder.State) {
+                renderState(holder, state)
+            }
+        }
+        voiceBroadcastRecorder?.addListener(recorderListener)
+        renderHeader(holder)
+    }
+
+    private fun renderHeader(holder: Holder) {
+        with(holder) {
+            roomItem?.let {
+                attributes.avatarRenderer.render(it, roomAvatarImageView)
+                titleText.text = it.displayName
+            }
+        }
+    }
+
+    private fun renderState(holder: Holder, state: VoiceBroadcastRecorder.State) {
+        with(holder) {
+            when (state) {
+                VoiceBroadcastRecorder.State.Recording -> {
+                    stopRecordButton.isEnabled = true
+
+                    liveIndicator.isVisible = true
+                    liveIndicator.tintBackground(colorProvider.getColorFromAttribute(R.attr.colorOnError))
+
+                    val drawableColor = colorProvider.getColorFromAttribute(R.attr.vctr_content_secondary)
+                    val drawable = drawableProvider.getDrawable(R.drawable.ic_play_pause_pause, drawableColor)
+                    recordButton.setImageDrawable(drawable)
+                    recordButton.contentDescription = holder.view.resources.getString(R.string.a11y_pause_voice_broadcast_record)
+                    recordButton.setOnClickListener { attributes.callback?.onTimelineItemAction(VoiceBroadcastAction.Recording.Pause) }
+                    stopRecordButton.setOnClickListener { attributes.callback?.onTimelineItemAction(VoiceBroadcastAction.Recording.Stop) }
```

Comment:
You can use `onClick {` which include a debouncer. Applicable to all `setOnClickListener` in the Item